subprojects/platform-native/src/testFixtures/groovy/org/gradle/nativeplatform/fixtures/AbstractInstalledToolChainIntegrationSpec.groovy

```diff
@@ -50,68 +50,72 @@ allprojects { p ->
         })
     }
 
-    def NativeInstallationFixture installation(Object installDir, OperatingSystem os = OperatingSystem.current()) {
+    String getSourceType() {
+        return null
+    }
```

Comment:
Shouldn't this be abstract instead? What is the purpose of this method here?

Reply:
AbstractInstalledToolChainIntegrationSpec uses that method dynamically as a default argument to some methods. I first tried making the method abstract, but there are some subclasses where it makes no sense to set a global source type. I ended up using the default implemenation here.
I agree that this is not nice.